src/D2L.Security.OAuth2/Keys/KeyManagementService.cs

```diff
@@ -51,7 +51,7 @@ OAuth2Configuration config
 
 			var now = m_clock.UtcNow;
 
-			if ( current == null || current.ValidTo <= now ) {
+			if ( current == null || ExpectedTimeOfNewUsableKey( current ) < now ) {

```

Comment:
Oh I guess we talked about adding a buffer here, one sec

Reply:
Done

Reply:
If we were comfortable storing the refresh time somewhere the TimeSpan value returned by RefreshKeyAsync has a bunch of stuff built into it:
- if we have no valid key retry in 10 seconds
- retry 1 minute after a new key should be available
- retry every minute if there's supposed to be a newer key available